File: 342_power_of_four.c
#include <stdio.h>
#include <stdbool.h>

bool isPowerOfFour(int n)
{
    if (n <= 0)
        return false;

    return (n & (n - 1)) == 0 && (n - 1) % 3 == 0;
}

int main()

{
    int res = isPowerOfFour(16);
    printf("%d", res);
    return 0;
}

File: 342. Power of Four.py
import math


class Solution:
    def isPowerOfFour(self, n: int) -> bool:
        if n <= 0:
            return False
        res = math.log(n, 4)
        return res.is_integer()


newIsPower = Solution()
print(newIsPower.isPowerOfFour(1))
